src/components/AuditLogsTable/AuditLogsTable.tsx

```diff
@@ -66,14 +71,16 @@ const AuditLogsTable = ({ showModel = false }: Props) => {
     if (!wsControllerURL || !hasControllerConnection) {
       return;
     }
-    setLoading(true);
-    promiseDispatch<AuditEvents>(
-      jujuActions.findAuditEvents({ wsControllerURL })
-    ).then((response) => {
-      setLoading(false);
-      setAuditLogs(response?.events);
-    });
-  }, [hasControllerConnection, promiseDispatch, wsControllerURL]);
+    dispatch(jujuActions.fetchAuditEvents({ wsControllerURL }));
+  }, [hasControllerConnection, dispatch, wsControllerURL]);
+
+  useEffect(
+    () => () => {
+      // Clear audit events when the component is unmounted.
+      dispatch(jujuActions.clearAuditEvents());
```

Comment:
There's no need to keep out of date audit event data in the store once we stop showing the table.